compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -6037,7 +6037,16 @@ private BType checkInvocationArgs(BLangInvocation iExpr, List<BType> paramTypes,
             // function call error.
             if (i == 0 && arg.typeChecked && iExpr.expr != null && iExpr.expr == arg) {
                 BType expectedType = paramTypes.get(i);
-                types.checkType(arg.pos, arg.getBType(), expectedType, DiagnosticErrorCode.INCOMPATIBLE_TYPES);
+                BType actualType = arg.getBType();
```

Comment:
@KRVPerera @MaryamZi I hope this fix is fine, as this is very specific to string:Char